targettests/execution/qubit_management_if-classical.cpp

```diff
@@ -0,0 +1,34 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2024 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+#include <cudaq.h>
+
+// RUN: nvq++ --target opt-test --target-option dep-analysis,qpp %s -o %t && %t
+
+struct run_test {
+  __qpu__ auto operator()() {
+    cudaq::qubit p;
+
+    if (true) {
+      rx(1., p);
+      z(p);
+      rx(1., p);
+    } else {
+      rx(1., p);
+      y(p);
+    }
+    auto res = mz(p);
+    return res;
+  }
+};
+
+int main() {
+  bool result = run_test{}();
+  printf("Result = %b\n", result);
+  return 0;
```

Comment:
Should this return non-zero if the value of result is incorrect?

Reply:
Sounds good. But noting that it looks like this command line doesn't actually check the JIT IR dump.

Reply:
The JIT IR can't currently be dumped by the target. I think checking the pass statistics would also be a good testing approach (and would be a little easier to set up with nvq++ and the custom target).

Reply:
Oh. Does setting `CUDAQ_DUMP_JIT_IR=1` in the environment not work?

Reply:
The current approach does not go through JIT. Will create a separate issue for this one (or add to a new grouped issue). The better approach would be to use statistics here, and check the statistics.

Reply:
https://github.com/NVIDIA/cuda-quantum/issues/2181 written. Resolving this conversation.